features/cloud.py

```diff
@@ -481,6 +496,97 @@ def _create_instance(
+
+    :param gce_credentials_path
+        The GCE credentials path to use when authentiacting to GCE
+    :param gce_project
+        The name of the GCE project to be used
+    :machine_type:
+        A string representing the type of machine to launch (pro or generic)
+    :region:
+        The region to create the resources on
+    :tag:
+        A tag to be used when creating the resources on the cloud provider
+    :timestamp_suffix:
+        Boolean set true to direct pycloudlib to append a timestamp to the end
+        of the provided tag.
+    """
+
+    env_vars: "Tuple[str, ...]" = ("gce_credentials_path", "gce_project")
+
+    def __init__(
+        self,
+        machine_type: str,
+        region: "Optional[str]" = "us-west2",
+        tag: "Optional[str]" = None,
+        timestamp_suffix: bool = True,
+        zone: "Optional[str]" = "a",
+        gce_credentials_path: "Optional[str]" = None,
+        gce_project: "Optional[str]" = None,
+    ) -> None:
+        self.gce_credentials_path = gce_credentials_path
+        self.gce_project = gce_project
+        self.zone = zone
+        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = str(
+            self.gce_credentials_path
+        )
+
+        super().__init__(
+            machine_type=machine_type,
+            region=region,
+            tag=tag,
+            timestamp_suffix=timestamp_suffix,
+        )
+
+    @property
+    def api(self) -> pycloudlib.cloud.BaseCloud:
+        """Return the api used to interact with the cloud provider."""
+        if self._api is None:
+            self._api = pycloudlib.GCE(
+                tag=self.tag,
+                timestamp_suffix=self.timestamp_suffix,
+                project=self.gce_project,
+                zone=self.zone,
+                region=self.region,
+            )
+
+        return self._api
+
+    def _create_instance(
+        self,
+        series: str,
+        instance_name: "Optional[str]" = None,
+        image_name: "Optional[str]" = None,
+        user_data: "Optional[str]" = None,
+    ) -> pycloudlib.instance:
+        """Launch an instance on the cloud provider.
+
+        :param series:
+            The ubuntu release to be used when creating an instance. We will
+            create an image based on this value if the used does not provide
+            a image_name value
+        :param instance_name:
+            The name of the instance to be created
+        :param image_name:
+            The name of the image to be used when creating the instance
+        :param user_data:
+            The user data to be passed when creating the instance
+
+        :returns:
+            An AWS cloud provider instance
+        """
+        if not image_name:
+            image_name = self.locate_image_name(series)
+
+        print("--- Launching GCE image {}({})".format(image_name, series))
+        inst = self.api.launch(image_id=image_name, user_data=user_data)
+        return inst
+
+
```

Comment:
Please add a 
```suggestion

    @property
    def pycloudlib_cls(self):
        """Return the pycloudlib cls to be used as an api."""
        return pycloudlib.GCE

```

Reply:
Once we have that api property refactoring, I do believe we will need this method, but right now, I don't think it will be adding a lot of value